File: Python/Problems/Problems/problem3423.py
from .problem import Problem

class Problem3423(Problem):
    def test(self):
        s = [ 1, 2, 4]
        expected = 3

        result = self.maxAdjacentDistance(s)

        print(result)

        return expected == result


    def maxAdjacentDistance(self, nums):
        maxDiff = abs(nums[len(nums) - 1] - nums[0])

        for i in range(len(nums) - 1):
            diff = abs(nums[i] - nums[i + 1])
            if (diff > maxDiff):
                maxDiff = diff

        return maxDiff

File: C++/Problems/Problems/problem_3423.cpp
#include "problem_3423.h"
#include <cmath>

bool problem_3423::test() {
    std::vector<int> nums =  { 1, 2, 4 };


    auto expected = 3;

    auto result = maxAdjacentDistance(nums);

    return expected == result;
}

int problem_3423::maxAdjacentDistance(std::vector<int> &nums) {
    auto maxDiff = std::abs(nums[nums.size() - 1] - nums[0]);

    for (int i = 0; i < nums.size() - 1; i++)
    {
        auto diff = std::abs(nums[i] - nums[i + 1]);
        if (diff > maxDiff)
            maxDiff = diff;
    }

    return maxDiff;
}
